SparkPost/matchbox
# UX-608 Setup typescript in matchbox

### What Changed
- Sets up typescript in `matchbox` and fixes setup in `matchbox-icons`

### How To Test or Verify
- Visit the matchbox package and run the build `npm run build`, or run `npm run build` in the root
- Verify `dist` is populated correctly with types and other rollup output

### PR Checklist
